pkg/maintainer/spv/spv.go

```diff
@@ -57,10 +105,348 @@ func (sm *spvMaintainer) startControlLoop(ctx context.Context) {
 }
 
 func (sm *spvMaintainer) maintainSpv(ctx context.Context) error {
-	logger.Infof("Maintaining SPV proof...")
+	for {
+		if err := sm.proveDepositSweepTransactions(); err != nil {
+			return fmt.Errorf(
+				"error while proving deposit sweep transactions: [%v]",
+				err,
+			)
+		}
 
-	// TODO: Implement. For now, just wait.
-	<-ctx.Done()
+		// TODO: Add proving of other type of SPV transactions: redemption
+		// transactions, moving funds transaction, etc.
+
+		select {
+		case <-time.After(sm.config.IdleBackOffTime):
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+}
+
+func (sm *spvMaintainer) proveDepositSweepTransactions() error {
+	depositSweepTransactions, err := sm.getUnprovenDepositSweepTransactions()
+	if err != nil {
+		return fmt.Errorf(
+			"failed to get unproven deposit sweep transactions: [%v]",
+			err,
+		)
+	}
+
+	txProofDifficultyFactor, err := sm.spvChain.TxProofDifficultyFactor()
+	if err != nil {
+		return fmt.Errorf(
+			"failed to get transaction proof difficulty factor: [%v]",
+			err,
+		)
+	}
+
+	// TODO: Handle a situation in which the block headers in the proof span
+	//       multiple Bitcoin difficulty epochs.
+	requiredConfirmations := uint(txProofDifficultyFactor.Uint64())
+
+	for _, transaction := range depositSweepTransactions {
+		accumulatedConfirmations, err := sm.btcChain.GetTransactionConfirmations(
+			transaction.Hash(),
+		)
+		if err != nil {
+			return fmt.Errorf(
+				"failed to get transaction confirmations: [%v]",
+				err,
+			)
+		}
+
+		if accumulatedConfirmations < requiredConfirmations {
+			// Skip the transaction as it has not accumulated enough
+			// confirmations. It will be proven later.
+			continue
```

Comment:
Let's add an info-level log for this case.

Reply:
Done in 132ac7d666c8964b47a71255b3c50334f240fe22.